src/cache.ts

```diff
@@ -0,0 +1,114 @@
+import { KubernetesObject } from './types';
+import { Watch } from './watch';
+
+export interface ObjectCache<T> {
+    get(name: string, namespace?: string): T | null;
+    list(namespace?: string): T[];
+}
+
+export type ListCallback<T extends KubernetesObject> = (list: T[]) => void;
+
+export class ListWatch<T extends KubernetesObject> implements ObjectCache<T> {
+    private objects: T[] = [];
+    private indexCache: any = {};
+    private path: string;
```

Comment:
I think these can be readonly, and that you can remove the explicit declarations and use TypeScript's declare-and-capture-in-the-constructor syntax `constructor(private readonly path: string, ...) { ... }`.

Reply:
done.